### ISSUE TITLE
some suggestions

### ISSUE THREAD

**Issue by sixsixfive** [tsujan/Kvantum#70]
just some ideas i had after updating some qss stuff

#1 colorize with qpalette  

now thats something that would be needed for proper color scheme support

replace actual svg colors with the current qpalette colors

eg: in svg

```
fill="#ff00ff"
```
becomes:
```
fill="palette(highlight)"
```

#2 allow subthemes(parents & childs)

eg. like in css/qss

QToolbar > QLineEdit

this would allow people to use different buttons everywhere eg a different lineedit, checkbox, whatever in toolbars or to make windows white but dialogs black(eg bespin),(I know its already possible but but it wouldnt make sense since both windows and dialogs share the same widgets)

not sure if it can be done, but it would be a huge improvement. and would give Kvantum almost the same theme abilites as GTK2 had

**Comment by tsujan** [comment]
> replace actual svg colors with the current qpalette colors..

No! For obvious reasons ;)

> now thats something that would be needed for proper color scheme support

Color scheme is a KDE concept. It is *not* equal to QPalette. A color scheme is a set of settings KDE provides for its own theme engine (Breeze or Oxygen). Kvantum doesn't depend on KDE but just because KDE is an important DE and some KDE apps use those schemes specifically, I've included them in installation.

> allow subthemes(parents & childs)... QToolbar > QLineEdit

I didn't get this part at all. What do you mean by `QToolbar > QLineEdit` when QToolbar and QLineEdit are totally different QWidgets?

As for black dialogs and white windows, you already know it's possible and I didn't understand the rest of that paragraph.

> would give Kvantum almost the same theme abilites as GTK2 had...

Which abilities?

Although I'll always welcome your reports and encourage you to continue them, your recent reports haven't been as productive as the previous ones. Please *be more specific* (as you were before)	and forget about *your own taste* here (-> #69)! Taste based statements could be useful as comments on artistic works like themes but not about codes.

**Comment by sixsixfive** [comment]
>Color scheme is a KDE concept. 

there are many people now that use qt style sheets to set the qt colors(afaik qtct uses them too)

eg https://gist.github.com/QuantumCD/6245215

 >didn't get this part at all. What do you mean by...

http://www.w3.org/TR/CSS21/selector.html#child-selectors
http://doc.qt.io/qt-4.8/stylesheet-syntax.html#selector-types

you know what i mean i saw your css/qss lxqt theme in kvsimple(eg Plugin > QToolButton)

>As for black dialogs and white windows, you already know it's possible and I didn't understand the rest of that paragraph.

eg I have a grey window the button is probably with black text, now i make only the dialogs black i would have a black dialog with a grey button and black text

>your recent reports haven't been as productive as the previous ones.

thats just because kvantum is now almost bug-free

>your own taste here (-> #69)!

I just added it because you said it would ok, but for me it looks broken like all the old GTK2 themes back then that tried to mimic osx

**Comment by tsujan** [comment]
Oh, those are Qt stylesheets!
Kvantum wasn't supposed to be based on stylesheets from the start but on a single SVG image.

If I chose a stylesheet structure for Kvantum, I should accept the limitations which would come inevitably. QtCurve was the best cadidate for having a stylesheet structure and at first, I thought about forking it but I wanted Kvantum to be as free and flexible as possible. I knew that everything had its cost: I would lose the handy aspects of stylsheets. The parent-child relationship you mentioned was one of them. However, I devised my own ways of checking for containment relationships that go beyong a simple parent-child one.

Now I don't have enough time to enumerate the limitations of stylesheets and how Kvantum goes beyond them but to me, they've been coding "facts". Also a glance at how visually different themes are made for Kvantum and how the themes of other style engines are similar to each other might be enough.

All in all, a stylesheet theme engine is a totally different project, which I'm not interested in.

> I have a grey window the button is probably with black text...

Kvantum is the first Qt theme engine that allows for dark-and-light themes (http://kde-look.org/content/show.php/KvHybrid+for+Kvantum?content=168699). However, it also has its limitations, which I've explained in "Theme-Making.pdf" -> Flat Indicators and High Contrast:

```
Therefore, apart from textless widgets, only the interior elements of those widgets that accept state- specific text colors can have a high contrast with the window or base background...
```
